example/mapreduce/task_interface_mapreduce.go

```diff
@@ -0,0 +1,19 @@
+package mapreduce
+
+import (
+	"github.com/taskgraph/taskgraph"
+)
+
+// MapreduceTask is a logic repersentation of a computing unit in mapreduce framework.
+// Each task contains at least one Node.
+
+type MapreduceTask interface {
+	// task interface of mapreduce framework
+	taskgraph.Task
+	// For mapper work, after processing by function of user
+	// it need provide user mapper function a method to emit their result to shuffle,
+	// Thus framework could take these data to specfic shuffle.
+	Emit(string, string)
```

Comment:
Can you clarify these two functions and what they are used for? Adding examples in comments is better.